tests/installation/grub_test.pm

```diff
@@ -88,6 +88,8 @@ sub run {
         # avoid timeout for booting to HDD
         send_key 'ret';
     }
+    # Avoid return key not received occasionally for hyperv-uefi guest at first boot
+    send_key 'ret' if (check_var('HYPERV_VERSION', '2012r2') && check_var('UEFI', '1'));
```

Comment:
With one more time send_key, I am afraid the survival chance will not get significant improvement. What about changing line 89 to send_key_until_needlematch?

Reply:
If using send_key_until_needlematch, we could not assert the needle match since it's not related to grub_test. It will lead the test module grub_test failed in case the statement fails. Another way is to make the changes in first_boot test module, refer to the commit 4e070ec.

Reply:
I do not look in depth for the whole story of the test. But if commit 4e070ec works to make test more stable, good to me.